cmd/fyne/internal/commands/package.go

```diff
@@ -412,6 +421,35 @@ func mergeMetadata(p *appData, data *metadata.FyneApp) {
 	}
 }
 
+// normaliseIcon takes a non-png image file and converts it to PNG for use in packaging.
+// Successful conversion will return a path to the new file.
+// Any errors that occur will be returned with an empty string for new path.
+func normaliseIcon(path string) (string, error) {
+	// convert icon
+	img, err := os.Open(path)
+	if err != nil {
+		return "", fmt.Errorf("failed to open source image: %w", err)
+	}
+	defer img.Close()
+	srcImg, _, err := image.Decode(img)
+	if err != nil {
+		return "", fmt.Errorf("failed to decode source image: %w", err)
+	}
+
+	tmpPath := filepath.Join(os.TempDir(), "fyne-ico-tmp.png")
+	out, err := os.Create(tmpPath)
```

Comment:
Would it be possible to remove the file once it is not necessary anymore?

Reply:
This is a good idea.
I don't think we have a hook for the processed completing at this time.
Can you think of a clean way to do that?

Reply:
doPackage use `defer p.removeBuild(files)`. Maybe make doPackage and validate return a `[]string` of files that need to be cleaned and set a defer in the caller to those function?

Reply:
We could create a folder in the temp directory using https://pkg.go.dev/io/ioutil#ReadDir and save everything there when we create temporary files. We then simply make sure that we just remove the whole folder before exiting the command (we could probably even hook up the interup signal https://www.tutorialspoint.com/how-to-handle-signals-in-golang so that it removes it before exiting using ctrl+c).